packages/desktop-client/src/components/App.js

```diff
@@ -28,6 +29,7 @@ class App extends Component {
   };
 
   async init() {
+    await clearServer();
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏why do we need this here? I'm worried about what other unexpected knock-on effect it might have.

Reply:
Maybe it makes more sense to put this a few lines down right above initConnection.

I struggled to really figure out a good way to make 100% sure that I close the old socket, before opening the new one. 

I initially put it in a 'unload' event (or similar), but that wouldn't for some reason always guarantee it had run before init was called for the reload. Maybe someone with more react knowledge can let me know a better strategy i'd be happy to change it. 

That being said, I couldn't really envisage a scenario where this would be problematic, as we always want to ensure that the old socket is closed before loading the new one, and in the browser this function does nothing so won't impact the web workers.

Reply:
What if we moved this `clearServer` call inside [here](https://github.com/actualbudget/actual/blob/master/packages/loot-core/src/platform/client/fetch/index.web.ts#L81)? In that way we isolate it this logic to only the socket code.

Or better yet: add a conditional to the `init` function. If the socket is already active - don't re-initialise and just re-use the active one (might not be possible though, but throwing it out as an idea).

Reply:
I made this change. It does however always re-create the socket for a couple reasons which may not be valid & i'm happy to try the other approach so let me know.

- There may be a scenario where you intentionally want to reload the socket due to some network issue?
- This is more in line with how the web version works, which i believe re-creates the web worker
- If for whatever reason the `socketName` changes in the call to `init` you end up needing more complex logic to decide if you should re-create or not.

Reply:
Very good points! But I think we should try to focus on the current problem at hand and not over-engineer this for a future that might not come. Lets try to keep things simple.

Reply:
Just to clarify, is this comment in support of the current solution, or a solution that keeps the socket open? Less code logic needed the current way, but happy to change!

Reply:
The current solution is good. We can keep as-is :)